File: Prácticas/P1/Java/civitas/src/main/java/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.Random;

/**
 *
 * @author Jose Padial
 */
public class Dado {
    private static final Dado INSTANCE = new Dado();
    private static final int SalidaCarcel = 5;
    
    private Random random;
    private int ultimoResultado;
    private boolean debug = false;
    
    private Dado(){}
    
    static Dado getInstance(){
        return INSTANCE;
    }
    
    int tirar(){
        if(debug == false){
            random = new Random();
            ultimoResultado = 1 + random.nextInt(6);
            return ultimoResultado;
        }
        return ultimoResultado=1;
    }
    
    boolean salgoDeLaCarcel(){
        return tirar() == SalidaCarcel;
    }
    
    int quienEmpieza(int n){
        random = new Random();
        return random.nextInt(n);
    }

    void setDebug(boolean d) {
        this.debug = d;
    }

    int getUltimoResultado() {
        return ultimoResultado;
    }      
}

File: Prácticas/P4/RUBY/Civitas/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "diario.rb"
require 'singleton'

module Civitas
class Dado
  
  include Singleton
  SALIDACARCEL = 5
  
  def initialize
    @ultimoResultado
    @debug = false
  end
  
  public 
  
  def tirar
    if(@debug == false)
      resultado = Random.rand(1..6)
      return 3
    else
      return 1
    end
  end
  
  def salgo_de_la_carcel
      if(tirar == SALIDACARCEL)
        return true
      else
        return false
      end
      
  end
  
  def quien_empieza(n)
    return Random.rand(1..n)
  end
  
  def set_debug(deb)
    @debug = deb
    diario = diario.instance
    diario.ocurre_evento("se ha cambiado el estado de debug a #{@debug}")
  end
  
  def get_ultimo_resultado
    @ultimoResultado
  end
  
  
end
end
